python/lsst/rapid/analysis/quickLook.py

```diff
@@ -0,0 +1,281 @@
+# This file is part of rapid_analysis.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from lsst.ip.isr import IsrTask
+from lsst.ip.isr.isrTask import IsrTaskConnections
+import lsst.pipe.base.connectionTypes as cT
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from lsst.pipe.tasks.characterizeImage import CharacterizeImageTask
+from lsst.meas.algorithms.installGaussianPsf import InstallGaussianPsfTask
+
+__all__ = ['QuickLookTask', 'QuickLookTaskConfig']
+
+
+def _getArgs(connection):
+    """Get the all the required args from a connection in order to reconstruct.
+    """
+    newArgs = {}
+    for attr in ("name", "storageClass", "multiple", "doc", "dimensions", "isCalibration", "deferLoad",
+                 "minimum", "lookupFunction"):
+        if hasattr(connection, attr):
+            newArgs[attr] = getattr(connection, attr)
+    return newArgs
+
+
+class QuickLookTaskConnections(IsrTaskConnections):
+    def __init__(self, *, config=None):
+        """Copy isrTask's connections, changing prereq min values to zero.
+
+        Copy all the connections directly for IsrTask, keeping ccdExposure as
+        required as non-zero, but changing all the other PrerequisiteInputs'
+        minimum values to zero.
+        """
+        super().__init__(config=IsrTask.ConfigClass())  # need a dummy config, isn't used other than for ctor
+        for name, connection in self.allConnections.items():
+            if hasattr(connection, 'minimum'):
+                args = _getArgs(connection)
```

Comment:
For sure the block doesn't smell great, but I wrote it with Nate L, and he said this was the best way to go about it. I don't think what you say about the `IsrTaskConnections` having `minimum=0` is true though; bias, dark, flat and defect are all missing that and instead have `isCalibration=True`.

Reply:
If you can show me a better way I'm very open to it, as I don't love this at all, but like I said, this was Nate's best guess, and I couldn't make it work otherwise.

Reply:
As we discussed in person, this at least needs to be clearly documented (both what the goal is, and what this code actually does) in this Connections class docstring. I'm still uncomfortable with it, but I don't know of a better approach, given your needs here.